### PR TITLE
Add LLVM Windows MSVC build/packaging/archive/download support.

### PR DESCRIPTION
**Pull request by lateralusX**
Several changes and improvements to hook into and support LLVM build/packaging/archive/download support for Windows MSVC build LLVM libraries.

Separate build support is added to LLVM repro in https://github.com/mono/llvm/pull/43. That will make sure we have on commit builds for Mono LLVM release60 branch, uploading artifacts to azure blob storage, that could be downloaded and installed as part of CI for Windows x64 full AOT + LLVM PR jobs (similar to how it works for our Linux full AOT + LLVM PR jobs).

PR also include support to build x64 MSVC LLVM version as part of sdks-llvm build, extending current make file system to include Windows. NOTE, this PR doesn't add support into run-jenksin.sh and no additional CI jobs will be setup at the moment building such an artifact, but all support is added into the build scripts, making it possible to build MSVC x64 LLVM archives from Mono repository without the need to do full configure, as done in run-jenkins sdks-llvm build.

All LLVM build scripts can be executed from Cygwin, WSL or regular Windows command prompt.

Improvements to helper build scripts running msbuild and MSVC build mono-sgen.exe from Cygwin/WSL/Command prompt. Reuse of VS environment setup logic for msbuild and msvc, prepared for optimization, not setting up full VS env on each AOT compile invocation as part of CI builds etc.

Added support to use external LLVM as part of MSVC build infrastructure. The LLVM build target was not executed when using external build LLVM, but rest was setup and configured as expected. PR add support to copy needed LLVM utilities into build output folder in case where we are using an external LLVM build.

### CODE REVIEW

**Comment by lateralusX** [comment]
Did a test using our Windows x64 FullAOT PR job to use pre-build LLVM downloaded and used in build and test. All test passed.

**Comment by lateralusX** [comment]
@monojenkins build Windows x64 FullAOT+LLVM

**Comment by lateralusX** [comment]
https://jenkins.mono-project.com/view/Pull%20requests/job/xy2/2/, pass running this PR against new Windows x64 FullAOT+LLVM using pre-build package.

**Comment by directhex** [comment]
I fixed your job to add a context (it's showing as 'default' as-is in the checks pane in GH)

File: sdks/builds/Makefile [hunk 1]
```diff
@@ -11,31 +11,58 @@ RELEASE=$(if $(filter $(CONFIGURATION),release),1)
...
+DISABLE_LIBTOOLIZE=1
+endif
+
+CCACHE:=$(if $(DISABLE_CCACHE),,$(shell which ccache))
+NINJA:=$(shell which ninja)
+
 include $(TOP)/sdks/versions.mk
 include $(TOP)/sdks/paths.mk
 
+ifndef DISABLE_LIBTOOLIZE
```
Comment (lambdageek) [inline sdks/builds/Makefile]: nit: this doesn't disable libtoolize. it just disables some path tweaks.

File: sdks/builds/Makefile [hunk 2]
```diff
@@ -11,31 +11,58 @@ RELEASE=$(if $(filter $(CONFIGURATION),release),1)
...
-
 UNAME=$(shell uname)
 
+ifneq (,$(findstring CYGWIN,$(UNAME)))
+UNAME=Windows
+endif
+
+ifeq ($(UNAME),Linux)
+UNAME_WSL_CHECK=$(shell uname -a)
+ifneq (,$(findstring Microsoft,$(UNAME_WSL_CHECK)))
```
Comment (jaykrell) [inline sdks/builds/Makefile]: Can this be different somehow? It breaks wasm on WSL. I don't know if it works, I'm trying it, but this definitely breaks it.

**Pull request merged**